compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -439,7 +439,9 @@ object TypeOps:
           val lo1 = apply(lo)
           if (lo1 ne lo) lo1 else tp
         case tp: LazyRef if isExpandingBounds =>
-          emptyRange
+          if variance == 1 then defn.AnyType
+          else if variance == -1 then defn.NothingType
+          else TypeBounds.empty
```

Comment:
This is still not correct conceptually -- we need to know both the _kind_ and _variance_ of the LazyRef in order to produce an approximation.